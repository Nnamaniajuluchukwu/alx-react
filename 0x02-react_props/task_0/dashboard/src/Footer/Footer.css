:root {
    --font-family-general: Arial, Helvetica, sans-serif;
}

.App-footer {
    text-align: center;
    padding: 1rem;
    font-style: italic;
    font-family: var(--font-family-general);
}
