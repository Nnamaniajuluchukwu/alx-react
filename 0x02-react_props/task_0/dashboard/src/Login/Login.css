
.App-body {
    font-size: 1rem;
    font-family: var(--font-family-general);
    margin: auto;
    height: 23rem;
    padding: 3rem;
}
  
.App-body input {
    margin-right: 5px;
}
