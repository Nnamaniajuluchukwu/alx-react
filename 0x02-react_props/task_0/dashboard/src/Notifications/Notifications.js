import React from "react";
import './Notifications.css';
import { getLatestNotification } from "../utils/utils";


export default function Notifications() {
    return (
        <div className="Notifications">
            <p>Here is the list of notifications</p>
            <ul>
                <li data-priority="default">New course available</li>
                <li data-priority="urgent">New resume available</li>
                <li dangerouslySetInnerHTML={{ __html: getLatestNotification() }}></li>
            </ul>
            <button style={{
                float: 'right',
                marginTop: '-9.5em',
                marginRight: '-0.5em',
                backgroundColor: '#fff',
            }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                <img src={require('../assets/close-button.png')} height="15px" width="13px" alt="close icon"/>
            </button>
        </div>
    );
}
