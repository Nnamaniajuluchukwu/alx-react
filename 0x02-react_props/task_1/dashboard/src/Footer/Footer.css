.App-footer {
    text-align: center;
    padding: 1rem;
    font-style: italic;
    font-family: var(--font-family-general);
}
