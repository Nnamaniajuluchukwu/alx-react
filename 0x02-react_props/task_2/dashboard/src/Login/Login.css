
.App-body {
    border-bottom: 3px solid #e0354b;
    font-size: 1rem;
    font-family: var(--font-family-general);
    margin: auto;
    height: 23rem;
    padding: 3rem;
}
  
.App-body input {
    margin-right: 5px;
}
