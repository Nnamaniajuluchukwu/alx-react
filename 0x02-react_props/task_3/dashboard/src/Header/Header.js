import React from "react";
import './Header.css';


export default function Header() {
    return (
        <header className="App-header">
            <img src={require('../assets/holberton_logo.jpg')} alt='holberton logo' />
            <h1>School dashboard</h1>
      </header>
    );
}
