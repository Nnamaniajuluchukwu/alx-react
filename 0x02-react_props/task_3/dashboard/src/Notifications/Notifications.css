.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
}

li[data-notification-type='default'] {
    color: blue;
}

li[data-notification-type='urgent'], li {
    color: red;
}
