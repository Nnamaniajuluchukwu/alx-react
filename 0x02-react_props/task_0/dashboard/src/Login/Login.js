import React from "react";
import './Login.css';


export default function Login() {
    return (
        <>
            <body className="App-body">
                <p>Login to access the full dashboard</p>
                <label for='email'>Email </label>
                <input type='email' id='email'></input>
                <label for='password'> Password </label>
                <input type='password' id='password'></input>
                <button type='submit'>OK</button>
            </body>
        </>
    );
}
