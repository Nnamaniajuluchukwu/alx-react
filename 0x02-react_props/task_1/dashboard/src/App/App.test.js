import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Notifications from '../Notifications/Notifications';
import Header from '../Header/Header';
import Login from '../Login/Login';
import Footer from '../Footer/Footer';


describe('App Component', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<App />);
    });

    it('should contain the Notifications component', () => {
      expect(wrapper.find(Notifications).length).toBe(1);
    });

    it('should contain the Header component', () => {
      expect(wrapper.find(Header).length).toBe(1);
    });

    it('should contain the Login component', () => {
      expect(wrapper.find(Login).length).toBe(1);
    });

    it('should contain the Footer component', () => {
      expect(wrapper.find(Footer).length).toBe(1);
    });

    it('should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
