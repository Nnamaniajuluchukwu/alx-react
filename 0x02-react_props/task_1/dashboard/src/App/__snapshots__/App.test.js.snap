// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
<Fragment>
  <Notifications />
  <div
    className="App"
  >
    <Header />
    <Login />
    <Footer />
  </div>
</Fragment>
`;
